addons/mail/static/src/components/composer/composer.js

```diff
@@ -192,6 +197,7 @@ class Composer extends Component {
         // TODO: we might need to remove trigger and use the store to wait for the post rpc to be done
         // task-2252858
         this.trigger('o-message-posted');
+        this._isPostingMessage = false;
```

Comment:
I'm not sure if we wouldn't want to debounce the event handlers instead?

But if we are going to do it this way, this might get stuck forever is the RPC crashes if we don't also handle reset flag on error.

Also this value looks important enough to be worth being a field. I would imagine that would be useful to actually disable the button while it will not have effect, could also serves as displaying a spinner some day if we want.

Reply:
Not sure debounce would be enough as is message_post is really slow for whatever reason and if the guy spam the enter/click button forever

Reply:
This will prevent people from sending multiple messages in quick succession though, people will probably complain.

On the other hand this looks like a direct fix to order issue of task-2073801

Reply:
Mmh normally the flag is reset when the input is reset, so normally you can send a message "quite" quickly after the other.
With a model field, the o-message-posted was still triggered so I went for the solution of rejecting the postMessage promise by throwing a specific Error. Maybe it could be done by making postMessage returning a status (like done/not done or the messageId but I'm not sure we get a messageId when it's a command)